src/shared/components/VideoPlayer/ControlsIndicatorManager.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from 'react'
+import { CSSTransition } from 'react-transition-group'
+import { VideoJsPlayer } from 'video.js'
+
+import {
+  SvgPlayerBackwardFiveSec,
+  SvgPlayerBackwardTenSec,
+  SvgPlayerForwardFiveSec,
+  SvgPlayerForwardTenSec,
+  SvgPlayerPause,
+  SvgPlayerPlay,
+  SvgPlayerSoundHalf,
+  SvgPlayerSoundOff,
+  SvgPlayerSoundOn,
+} from '@/shared/icons'
+
+import {
+  ControlsIndicator,
+  ControlsIndicatorTooltip,
+  ControlsIndicatorTransitions,
+  ControlsIndicatorWrapper,
+} from './ControlsIndicatorManager.style'
+import { CustomVideojsEvents, VOLUME_STEP } from './videoJsPlayer'
+
+import { Text } from '../Text'
+
+type VideoEvent = CustomVideojsEvents | null
+
+type EventState = {
+  type: VideoEvent
+  description: string | null
+  icon: React.ReactNode | null
+  isVisible: boolean
+}
+
+type ControlsIndicatorManagerProps = {
+  player: VideoJsPlayer | null
+  cachedPlayerVolume: number
+}
+
+export const ControlsIndicatorManager: React.FC<ControlsIndicatorManagerProps> = ({ player, cachedPlayerVolume }) => {
+  const [indicator, setIndicator] = useState<EventState | null>(null)
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const indicatorEvents = Object.values(CustomVideojsEvents)
+    const handler = (e: Event) => {
+      const playerVolume = e.type === CustomVideojsEvents.Unmuted ? cachedPlayerVolume || VOLUME_STEP : player.volume()
```

Comment:
Because I was getting the stale value from `player.volume()`. 
If I replace this with just `player.volume` and click `M` to mute video and then click `M` again to unmute, the indicator will show `0%` volume(but will show correct sound on icon). 
I could workaround this with setTimeout:
```tsx
  useEffect(() => {
    if (!player) {
      return
    }
    let timeout: NodeJS.Timeout
    const indicatorEvents = Object.values(CustomVideojsEvents)

    const handler = (e: Event) => {
      timeout = setTimeout(() => {
        const indicator = createIndicator(e.type as VideoEvent, player.volume(), player.muted())
        if (indicator) {
          setIndicator({ ...indicator, isVisible: true })
        }
      }, 0)
    }
    player.on(indicatorEvents, handler)

    return () => {
      clearTimeout(timeout)
      player.off(indicatorEvents, handler)
    }
  }, [player])
```

Let me know what you think